### PR TITLE
Bugfix: Fix deliberate cache misses causing massive logging on DB

### PR DESCRIPTION
**Pull request by coolcollo**
- Bug fix deliberate cache misses on checking if a flag is active
- Enabled caching of VerifiedUser and cache lookup of the same when checking if flag is active

### CODE REVIEW

File: waffle/models.py [hunk 1]
```diff
@@ -168,6 +170,23 @@ def uncache_switch(**kwargs):
...
+
+
+def cache_verified_user(**kwargs):
+    verified_user = kwargs.get('instance')
+    cache.add(keyfmt(settings.VERIFIED_USER_CACHE_KEY, verified_user.phone_number), verified_user)
+
+
+def uncache_verified_user(**kwargs):
+    verified_user = kwargs.get('instance')
+    cache.delete(keyfmt(settings.VERIFIED_USER_CACHE_KEY, verified_user.name))
```
Comment (solomonkaririjumo) [inline waffle/models.py]: Is phone number same as username. Or the keys for storage doesn't need to match the retrieval?

File: setup.py [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 setup(
     name='django-waffle-session',
-    version='0.2.4',
+    version='0.2.5',
```
Comment (komuw) [inline setup.py]: @coolcollo 

- hermes-pesa currently uses `django-waffle-session` version `0.2.2`; https://github.com/jumo/hermes-pesa/blob/f630a1bb5da5379c29048af04eb3ab5098a1b90f/requirements/base.txt#L32    
- however, gemfury doesn't have such a version;
Screenshot 2019-09-23 at 12 58 19
- pypi has that exact version; 
Screenshot 2019-09-23 at 12 59 25

So this leads me to believe H-pesa is still using Mwass' fork. 

We need to make sure that this PR is not branching from the master branch of `jumo/django-waffle-session` but instead it should branch from `version 0.2.2` of Mwass' fork. 
That way we do not risk introducing unwanted changes that are in `version 0.2.2` or omitting some features that are in `version 0.2.2`

Comment (komuw) [reply setup.py]: We could take whatever code that is `version 0.2.2` in Mwass' fork and apply them to the master branch of `jumo/django-waffle-session` and then start this PR from there

File: waffle/models.py (hunk 1 above)
Comment (coolcollo) [reply waffle/models.py]: @solomonkaririjumo Good catch, i don't even know where i got the `name` from. Must have been a natural tendency as a result of subject model name 🤦‍♂

File: setup.py (hunk 2 above)
Comment (coolcollo) [reply setup.py]: @komuw Good point, let me take a look and compare the two versions

**Comment by coolcollo** [comment]
> @coolcollo is it possible to have a testcase for the change?

@urandu I looked at the test situation of the repo in circleci and found it to be in tatters hence why i didn't bother adding more tests since they don't run to begin with. I figured out instead of wasting more time trying to fix them (root cause looks like outdated code), i just bug fix this one thing and put in a recommendation to refork from current version of django-waffle and fix everything from there. Nevertheless, i will try fixing the tests once more, if am unsuccessful i will fallback to my original plan.

**Comment by coolcollo** [comment]
> > @coolcollo is it possible to have a testcase for the change?
> 
> @urandu I looked at the test situation of the repo in circleci and found it to be in tatters hence why i didn't bother adding more tests since they don't run to begin with. I figured out instead of wasting more time trying to fix them (root cause looks like outdated code), i just bug fix this one thing and put in a recommendation to refork from current version of django-waffle and fix everything from there. Nevertheless, i will try fixing the tests once more, if am unsuccessful i will fallback to my original plan.

@urandu Addressed in this [commit](https://github.com/jumo/django-waffle-session/pull/4/commits/d2a4fc21d88034e1989bf6e005e352cf5b203cc6)

File: waffle/models.py (hunk 1 above)
Comment (coolcollo) [reply waffle/models.py]: Fixed in this [commit](https://github.com/jumo/django-waffle-session/pull/4/commits/d2a4fc21d88034e1989bf6e005e352cf5b203cc6)

File: setup.py (hunk 2 above)
Comment (coolcollo) [reply setup.py]: @komuw I dug into the versions conundrum and determined 0.2.2 was actually generated and uploaded to pypi from this repo, incidentally, by you 😆 
Screenshot 2019-09-25 at 11 54 38

The last version on Mwaas' fork was 0.2.1 as seen on the latest commit [here](https://github.com/mwaaas/django-waffle-session/commit/43ba35bd487ff7441535a920adf7856118615d3a)

Therefore the only changes we risk introducing that are not in production are changes from Muchina which were to add some resources to admin models as seen [here](https://github.com/jumo/django-waffle-session/commit/e983e7d125a4d64dbb411a8f7d937a858f152c4b) In my opinion, these changes shouldn't be breaking and in any case, we will validate on staging before deploy that the admin models functionality is not broken. Incidentally the double version bump-up (0.2.2 to 0.2.4) was effected during these changes. What is your opinion on this?

**Comment by komuw** [comment]
> The library setup on circleci is fatally broken. 

@coolcollo remember to add a jira card for that

File: setup.py (hunk 2 above)
Comment (4jah254) [reply setup.py]: Actually these changes from Muchina were discarded. While we were in Ninjas, we never implemented it. 
@urandu Kindly confirm if this is correct.

Comment (coolcollo) [reply setup.py]: @4jah254 @urandu Should I then revert the changes from the repo ?

**Comment by coolcollo** [comment]
> > The library setup on circleci is fatally broken.
> 
> @coolcollo remember to add a jira card for that

@komuw I had thought of an even drastic suggestion, we add a card to update this repo to be based on some current version of `django-waffle`. Our current version is based on some 2015 version which gives me jitters.

**Comment by 4jah254** [comment]
> > > The library setup on circleci is fatally broken.
> > 
> > 
> > @coolcollo remember to add a jira card for that
> 
> @komuw I had thought of an even drastic suggestion, we add a card to update this repo to be based on some current version of `django-waffle`. Our current version is based on some 2015 version which gives me jitters.

In agreement with updating.

**Pull request merged**